src/App.jsx

```diff
@@ -54,19 +54,43 @@ function Todos() {
             'https://raw.githubusercontent.com/IdoGvili/TaskList-repo/master/src/Data.json',
         );
         const fetchedData = await response.json();
-        const newData = fetchedData.map((task) => ({ ...task, id: v4() }));
+        const newData = fetchedData.map((toDo) => ({ ...toDo, id: v4() }));
         setToDoList(newData);
     }, []);
     useEffect(() => {
         fetchData();
     }, [fetchData]);
-
+    const handleSearchTerm = useCallback((searchTermProp) => {
```

Comment:
same here about the naming.

let's name this function `filterTodos` or `searchTodos` and have it responsible for the filtering. it should update your `toDoList` state which in turn will get passed on to `ToDoList` component